src/mpid/common/shm/mpidu_shm_alloc.c

```diff
@@ -0,0 +1,664 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/* vim: set ft=c.mpich : */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#include <mpidimpl.h>
+#include "mpl_shm.h"
+#include "mpidu_shm.h"
+#include "mpidu_shm_seg.h"
+
+#include <stdlib.h>
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#include <errno.h>
+
+#if defined (HAVE_SYSV_SHARED_MEM)
+#include <sys/ipc.h>
+#include <sys/shm.h>
+#endif
+
+#include <stddef.h>
+
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_SHM_RANDOM_ADDR_RETRY
+      category    : MEMORY
+      type        : int
+      default     : 100
+      class       : none
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_GROUP
+      description : >-
+        The default number of retries for generating a random address. A retrying
+        involves only local operations.
+
+    - name        : MPIR_CVAR_SHM_SYMHEAP_RETRY
+      category    : MEMORY
+      type        : int
+      default     : 100
+      class       : none
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_GROUP
+      description : >-
+        The default number of retries for allocating a symmetric heap in shared
+        memory. A retrying involves collective communication over the group in
+        the shared memory.
+=== END_MPI_T_CVAR_INFO_BLOCK ===
+*/
+
+enum {
+    MPIDU_SYMSHM_SUCCESS,
+    MPIDU_SYMSHM_MAP_FAIL,      /* mapping failure when specified with a fixed start addr */
+    MPIDU_SYMSHM_OTHER_FAIL     /* other failure reported by MPL shm */
+};
```

Comment:
The error codes are used only in this c file. Maybe we do not need `MPIDU_` prefix at all.